index.js

```diff
@@ -1943,6 +1943,29 @@
         return either.isRight ? [either.value] : [];
       }));
 
+  //# toEither :: a -> b? -> Either a b
+  //.
+  //. Takes a failure value, 'x', and a possibly null or undefined value, 'y'.
+  //. Returns a Right of 'y' if it is defined and not null, otherwise returns
+  //. a Left of 'x'.
+  //.
+  //. See also ['encaseEither'](#encaseEither)
+  //.
+  //. ```javascript
+  //. > S.toEither("No string", "ftp://example.com/")
+  //. Right("ftp://example.com/")
+  //.
+  //. > S.toEither("Invalid protocol", "ftp://example.com/".match(/^https?:\/\//))
+  //. Left("Invalid protocol")
+  //. ```
```

Comment:
Thanks for the explanation. I understand the problem now. I agree that it would be nice to use `String#match` (or `RegExp#exec`). Rather than use `[].slice.call` we could use `R.head`:

```javascript
> R.map(R.head, S.toEither('Invalid protocol', 'ftp://example.com/'.match(/^https?:/)))
Left('Invalid protocol')

> R.map(R.head, S.toEither('Invalid protocol', 'http://example.com/'.match(/^https?:/)))
Right('http:')
```

It's a bit longer than I would like, but it's quite a nice demonstration of the function. Perhaps we should precede these examples with two trivial examples.